aspnetcore/fundamentals/hosted-services/sample/Pages/Index.cshtml.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+using BackgroundTasksSample.Services;
+using System;
+using Microsoft.AspNetCore.Hosting;
+
+namespace BackgroundTasksSample.Pages
+{
+    public class IndexModel : PageModel
+    {
+        private readonly IApplicationLifetime _appLifetime;
+
+        #region snippet1
+        public IndexModel(IBackgroundTaskQueue queue, IApplicationLifetime appLifetime)
+        {
+            Queue = queue ?? throw new ArgumentNullException(nameof(queue));
+            _appLifetime = appLifetime;
+        }
+
+        public IBackgroundTaskQueue Queue { get; }
+        #endregion
+
+        public void OnGet()
+        {
+        }
+
+        #region snippet2
+        public IActionResult OnPostAddTask()
+        {
+            Queue.QueueBackgroundWorkItem(async token =>
+            {
+                var guid = Guid.NewGuid().ToString();
+
+                for (int delayLoop = 0; delayLoop < 3; delayLoop++)
+                {
+                    Console.WriteLine(
+                        $"{DateTime.UtcNow} - Queued Background Task {guid} is running. " +
+                        $"{delayLoop}/3");
+                    await Task.Delay(TimeSpan.FromSeconds(5), token);
+                }
+
+                Console.WriteLine(
+                    $"{DateTime.UtcNow} - Queued Background Task {guid} is complete. 3/3");
+            });
+
+            return RedirectToPage();
+        }
+        #endregion
+
+        public void OnPostShutdown()
+        {
+            _appLifetime.StopApplication();
```

Comment:
I wouldn't do this even in a sample, it's basically a way for somebody to shut your site down from a remote client. We can just describe that on exit the hosted service will stop.

Reply:
Just wanted to demo the `StopAsync` messages so the dev can :eye: see them.

Reply:
People copy and paste code from docs into their applications.

Reply:
The comment is supposed to head that off, but ok ... I'll pull it (here and over in Logger Message).

Reply:
Sure ... I'll pull it.